### ISSUE TITLE
Have a factory method Resource.fromClasspath(p: String)

### ISSUE THREAD

**Issue by wspringer** [jesseeichar/scala-io#12]
I find myself repeatedly doing things like this:

```
def fromClasspath(name: String, cl: Class[_] = ...)
  (implicit extraCloser: CloseAction[InputStream] = Noop) :     InputStreamResource[InputStream]=
  {
    val url = cl.getClassLoader.getResource(name)
    require(url != null)
    Resource.fromURL(url)(extraCloser)
  }
```

Having something like that as part of the Resource object would be awesome.

**Comment by jesseeichar** [comment]
I agree.  Although I think the API needs some refinement.  For example how to do you specify the classpath.  Sometimes the classloader from a particular class is good sometime the context classloader or perhaps a URLClassloader.

I think I will modify signature to something like:

def fromClasspath(name: String)(implicit classloader:Classloader)  the default classloader is a bit tricky.  Probably the context classloader.  But it would be nice to easily pass in a class, perhaps we need 2 fromClasspath methods.

**Comment by jesseeichar** [comment]
My idea was terrible :),  I am adding the method you provided

**Comment by jesseeichar** [comment]
Fixed as of 6533d07ac7cffd7567565289bf84bf60bb299c4b

**Comment by rtyley** [comment]
Just got a bit burnt by the preference for `cl.getClassLoader.getResource(name)` over just `cl.getResource()` here... I hadn't realised the behaviour of the the two was so different:

https://github.com/jesseeichar/scala-io/commit/6533d07ac7cffd7567565289bf84bf60bb299c4b#commitcomment-4687752
